endpoints/decorators/metrics.go

```diff
@@ -28,9 +28,9 @@ func (w *writerWithStatus) Header() http.Header {
 	return w.delegate.Header()
 }
 
-func MonitorHttp(handler httprouter.Handle, entry *metrics.MetricsEntry) httprouter.Handle {
+func MonitorHttp(handler httprouter.Handle, me *metrics.Metrics, method string) httprouter.Handle {
```

Comment:
I recommend replacing the string method with constants and using switch statements instead of string compares to offer better performance, which looks to me backtracking on a previous comment. Sigh. So, I think the combined approach here looks good, just replace the string compares with ints for a performance gain.

Reply:
done